src/openclon12/kernel_tasks.cpp

```diff
@@ -14,15 +14,25 @@
 
 #include "ImmediateContext.inl"
 
+#include "spookyv2.h"
+
 extern void SignBlob(void* pBlob, size_t size);
 constexpr uint32_t PrintfBufferSize = 1024 * 1024;
 constexpr uint32_t PrintfBufferInitialData[PrintfBufferSize / sizeof(uint32_t)] = { sizeof(uint32_t) * 2, PrintfBufferSize };
 
-auto Program::SpecializationKey::Allocate(D3DDevice const* Device, CompiledDxil::Configuration const& conf) -> std::unique_ptr<SpecializationKey>
+size_t Program::SpecializationKey::AllocatedByteSize(uint32_t NumArgs)
+{
+    uint32_t NumAllocatedArgs = NumArgs ? NumArgs - 1 : 0;

```

Comment:
I realize that this is just bringing forward what was already in the code, but why do we only allocate n-1 args? Shouldn't we have space allocated for all of them?

Reply:
Because the struct ends with an array of size 1, because 0-sized arrays are a compiler extension / unsupported in MSVC C++ I think.